cmake, epee: fix build with openssl v1.1 and with manually specified openssl path

radfish opened the pull request:
Two independent patches.

 epee: use boost type for SSL error code

Fixes compile error when building with OpenSSL v1.1:
```
contrib/epee/include/net/net_helper.h: In member function ‘void epee::net_utils::blocked_mode_client::shutdown_ssl()’:
contrib/epee/include/net/net_helper.h:579:106: error: ‘SSL_R_SHORT_READ’ was not declared in this scope
    if (ec.category() == boost::asio::error::get_ssl_category() && ec.value() != ERR_PACK(ERR_LIB_SSL, 0, SSL_R_SHORT_READ))
                                                                                                          ^
contrib/epee/include/net/net_helper.h:579:106: note: suggested alternative: ‘SSL_F_SSL_READ’
```
See boost/asio/ssl/error.hpp.
Boost handles differences between OpenSSL versions.


 cmake: epee: use var from FindOpenSSL.cmake

This fixes linking when path to openssl
is defined manually:
```
cmake -DOPENSSL_ROOT_DIR='/usr/include/openssl-1.0;/usr/lib/openssl-1.0' ...
```
This is useful for building with OpenSSL v1.0
when default system installation is v1.1.

The linking error is undefined SSL_load_error_strings symbol.
This is due to -L /usr/lib/openssl-1.0 not making it onto
the linkline (so -lssl pulls in the default system openssl).

MaxXor commented on contrib/epee/include/net/net_helper.h:
```diff
@@ -576,7 +576,7 @@ namespace net_utils
 			}
 			// Ignore "short read" error
-			if (ec.category() == boost::asio::error::get_ssl_category() && ec.value() != ERR_PACK(ERR_LIB_SSL, 0, SSL_R_SHORT_READ))
+			if (ec.category() == boost::asio::error::get_ssl_category() && ec.value() != boost::asio::ssl::error::stream_truncated)
```
This doesn't build with boost v1.60. `stream_truncated` was introduced in v1.62.

moneromooo-monero commented on contrib/epee/src/CMakeLists.txt:
```diff
@@ -49,6 +49,6 @@ target_link_libraries(epee
   PRIVATE
-    ssl
+    ${OPENSSL_LIBRARIES}
     crypto
```
You presumably want to remove crypto too then ?

radfish commented:
Thx for both. Removed crypto, and added ifdef for older boost + openssl 1.0.

Older boost won't link with openssl 1.1 anyway for other reasons, so there's no point in adding a third clause to the ifdef for old-boost-new-ssl. I added a check to CMake to preclude situations like #2667. 

I didn't test if 1.62 is the cutoff, but I'm going by what @MaxXor said.

MaxXor commented:
@radfish you can check it here:
http://www.boost.org/doc/libs/1_61_0/boost/asio/ssl/error.hpp boost v1.61 - `stream_truncated` not found
http://www.boost.org/doc/libs/1_62_0/boost/asio/ssl/error.hpp boost v1.62 - `stream_truncated` found

NanoAkron commented:
@radfish Would you also change the readme to reflect the change in the minimum version of boost?

MaxXor commented:
@NanoAkron That's not needed. It will use the old old when compiling with an older boost version.

anonimal commented on contrib/epee/include/net/net_helper.h:
```diff
@@ -576,7 +577,14 @@ namespace net_utils
 				m_io_service.run_one();
 			}
 			// Ignore "short read" error
```
I've not looked at the code but, is this at the end of session? If not, what good could come from ignoring a short read?
